### ISSUE TITLE
Look-in Received shows up twice. Why? 

### ISSUE THREAD

**Issue by akshayvernekar** [akshayvernekar/Brilliant_pad_issue_management#2]

**Comment by akshayvernekar** [comment]
This needs to be handled from App/server side , device sends the MQTT message only once but MQTT broker might send duplicate messages to receiving endpoints one or more times, app need to eliminate duplicate messages based on Packet ID.

**Comment by alanjcook** [comment]
Praveena : will. This be changed in next app?

**Comment by cumulations** [comment]
We need to understand the expected behavior here. Are we saying ignore two commands of the same type that comes from the device in a very close time interval?

What should be the time interval?

**Comment by alanjcook** [comment]
One lookin generates two lookins Serb and later two lookin received

One advance sheet generates two advance sheet sent windows. 

Generally there are twice as many windows as expected.

**Comment by akshayvernekar** [comment]
Hi @cumulations , the broker might send duplicate messages to the subscribers, please filter the events based on the packet ID or TS.

**Comment by alanjcook** [comment]
AKSHAY: Please confirm code is only sending one time. 

PRAVEENA: Will filtering below help resolve issue?

**Comment by akshayvernekar** [comment]
Yes we send only once , but app shows twice In pop up sometimes .

**Comment by alanjcook** [comment]
Has MQTT change fixed this?

**Comment by akshayvernekar** [comment]
We haven’t got the app with new MQTT settings yet . The changes needs to happen on both sides , firmware changes are ready .

**Comment by akshayvernekar** [comment]
Even with QOS1 broker might send the events one or more times , the real solution is in app side to reject duplicate packets based on packetID/ timestamp

**Comment by cumulations** [comment]
@alanjcook @akshayvernekar   let us differentiate between two cases where we get duplicate lookin events

Case 1
If the Event Log shows two Lookins with different size of images then this issue is not something that can be solved with QOS 1.

Case 2
If the App is showing 2 lookin popups but the event log shows only one, then that is something we can solve using QOS1.

Now, my question is are you seeing only Case 2 in your observations @alanjcook ?

**Comment by alanjcook** [comment]
Ah, yes, I am referring to case 2.

**Comment by akshayvernekar** [comment]
> If the App is showing 2 lookin popups but the event log shows only one, then that is something we can solve using QOS1
@cumulations  even with QOS1 you might get the events twice, QOS 1 will only guarantee that the event will be delivered 100% , you might still need to have a mechanism to filter the duplicates if at all thats the rootcause why multiple pop ups are shown.

**Comment by cumulations** [comment]
ok.. we will make necessary changes.
